src/h3_pyspark/indexing.py

```diff
@@ -33,14 +35,22 @@ def _index_line_object(line: LineString, resolution: int):
     Returns the set of H3 cells at the specified resolution which completely cover the input line.
     """
     result_set = set()
-    # Hexes for endpoints
-    endpoint_hexes = [h3.geo_to_h3(t[1], t[0], resolution) for t in list(line.coords)]
-    # Hexes for line (inclusive of endpoints)
-    for i in range(len(endpoint_hexes) - 1):
-        try:
-            result_set.update(h3.h3_line(endpoint_hexes[i], endpoint_hexes[i + 1]))
-        except h3.H3ValueError:
-            pass
+
+    # Hexes for vertices
+    vertex_hexes = [h3.geo_to_h3(t[1], t[0], resolution) for t in list(line.coords)]
+    result_set.update(vertex_hexes)
+
+    # Figure out the max-length line segment (step) we can process without interpolating
+    # https://github.com/kevinschaich/h3-pyspark/issues/8
+    endpoint_hex_edges = flatten(
+        [h3.get_h3_unidirectional_edges_from_hexagon(h) for h in [vertex_hexes[0], vertex_hexes[1]]]
```

Comment:
Shouldn't we iterate through all vertex_hexes

Reply:
Wait does a line assume its a line segment i.e. only two points? I thought a line geomerty can have multiple points